src/hotspot/share/gc/serial/serialHeap.hpp

```diff
@@ -35,6 +35,26 @@ class MemoryPool;
 class OopIterateClosure;
 class TenuredGeneration;
 
+// SerialHeap is the implementation of CollectedHeap for Serial GC.
+//
+// The heap is reserved up-front in a single contiguous block, split into two
+// parts, the young and old generation. The young generation resides at lower
+// addresses, the old generation at higher addresses. The boundary address
+// between the generations is fixed. Within a generation, committed memory
+// grows towards higher addresses.
+//
+//
+// low                                                                              high
+//
+//                                              +-- generation boundary (fixed after startup)
+//                                              |
+// |<-    young gen (reserved MaxNewSize)     ->|<- old gen (reserved MaxOldSize) ->|
+// +-----------------+--------+--------+--------+---------------+-------------------+
+// |       eden      |  from  |   to   |        |      old      |                   |
+// |                 |  (to)  | (from) |        |               |                   |
+// +-----------------+--------+--------+--------+---------------+-------------------+
```

Comment:
Unfortunately, the young gen contents are shown wrongly: from/to space are of fixed size and adjacent to the generation boundary.
There is no space between to-space and old gen.
Eden, from and to-space are committed separately within their space and never move.